wallet-new/src/Cardano/Wallet/Kernel/DB/TxMeta/Types.hs

```diff
@@ -171,3 +188,36 @@ instance Buildable TxMeta where
 
 instance Buildable [TxMeta] where
     build txMeta = bprint ("TxMetas: "%listJsonIndent 4) txMeta
+
+
+-- | Basic filtering & sorting types.
+
+newtype Offset = Offset { getOffset :: Integer }
+
+newtype Limit  = Limit  { getLimit  :: Integer }
+
+data SortDirection =
+      Ascending
+    | Descending
+
+data Sorting = Sorting {
+      sbCriteria  :: SortCriteria
+    , sbDirection :: SortDirection
+    }
+
+data SortCriteria =
+      SortByCreationAt
+    -- ^ Sort by the creation time of this 'Kernel.TxMeta'.
+    | SortByAmount
+    -- ^ Sort the 'TxMeta' by the amount of money they hold.
+
+-- | An opaque handle to the underlying storage, which can be easily instantiated
+-- to a more concrete implementation like a Sqlite database, or even a pure
+-- K-V store.
+data MetaDBHandle = MetaDBHandle {
+      closeMetaDB   :: IO ()
+    , migrateMetaDB :: IO ()
+    , getTxMeta     :: Core.TxId -> IO (Maybe TxMeta)
+    , putTxMeta     :: TxMeta -> IO ()
+    , getTxMetas    :: Offset -> Limit -> Maybe Sorting -> IO [TxMeta]
+    }
```

Comment:
so modular, wow,

the only other thing I'd say is "parameterize it over `m`" for ~*~pure tests~*~ but we can make that change later (eg via changing this to `data MetaDBHandleFor m = ...` and `type MetaDBHandle = MetaDBHandleFor IO`)

Reply:
hey, reviews aren't what's blocking this PR now :trollface: